modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -144,34 +136,36 @@ public void onFailure(Exception t) {
             protected void doRun() throws Exception {
                 client.performRequest(method, uri, params, entity, new ResponseListener() {
                     @Override
-                    public void onResponse(InputStream content) {
-                        T response;
+                    public void onSuccess(org.elasticsearch.client.Response response) {
+                        T parsedResponse;
                         try {
-                            XContent xContent = XContentFactory.xContentType(content).xContent();
-                            try(XContentParser xContentParser = xContent.createParser(content)) {
-                                response = parser.apply(xContentParser, () -> ParseFieldMatcher.STRICT);
+                            InputStream content = response.getEntity().getContent();
+                            XContentType xContentType = XContentType.fromMediaTypeOrFormat(
+                                    response.getEntity().getContentType().getValue());
```

Comment:
this is kinda important, I am not sure why we were auto-detecting content-type here. We rely on es to return the proper header. I have no fallback as es should always return the header. WDYT?